sensor_array/analysis/brute_force_analysis.py

```diff
@@ -641,6 +643,150 @@ def save_unbinned_array_pmf_data(gases, list_of_arrays, list_of_array_ids, all_a
+    color_map = matplotlib.cm.get_cmap(cmap)
+
+    data_dict_rgba = dict()
+    data_dict_zval = dict()
+    for i in range(len(a)):
+        atemp = (a[i]-min(a))/rescale_den
+        btemp = (b[i]-min(b))/rescale_den
+        ctemp = (c[i]-min(c))/rescale_den
+        A_out = np.round(atemp*100,3)
+        B_out = np.round(btemp*100,3)
+        C_out = np.round(ctemp*100,3)
+        rgba = color_map(color_norm(z[i]))
+        # data_dict[(bottom, right, left)] = ()
+        data_dict_rgba[(A_out, B_out, C_out)] = (rgba)
+        data_dict_zval[(A_out, B_out, C_out)] = z[i]
+
+    return(data_dict_rgba, data_dict_zval)
+
+def harper_ternary(data_dict, z, array_id, vmin, vmax, cmap, use_rgba, polygon_sf):
+    # Set image size and resolution
+    matplotlib.rcParams['figure.dpi'] = 1200
+    matplotlib.rcParams['figure.figsize'] = (3.25, 2.75)
+
+    # Initialize the Plot
+    scale = 100
+    figure, tax = ternary.figure(scale=scale)
+    tax.clear_matplotlib_ticks()
+    tax.get_axes().axis('off')
+    if use_rgba == True:
+        tax.heatmap(data_dict, style="h", use_rgba=use_rgba, colorbar=False, polygon_sf=polygon_sf)
+    elif use_rgba == False:
+        tax.heatmap(data_dict, style="dt", use_rgba=use_rgba, polygon_sf=polygon_sf)
+
+    # Set colorbar
+    # vmin = min(z)
+    # vmax = max(z)
+    color_norm = matplotlib.colors.Normalize(vmin=vmin, vmax=vmax)
+    color_map = matplotlib.cm.get_cmap(cmap)
+    sm = plt.cm.ScalarMappable(cmap=color_map, norm=color_norm)
+    sm.set_array([])
+    cbar = plt.colorbar(sm, ticks=np.linspace(0,vmax,6), fraction=0.02, format='%.1f')
+    cbar.ax.tick_params(labelsize=7)
+    # cbar.ax.set_ylabel(r'$Mass\ [mg/cm^3]$', rotation=90, fontsize=6)
+    cbar.ax.set_ylabel(r'Mass Uptake [$mg/cm^3$]', rotation=90, fontsize=9)
+
+    # Draw boundary, gridlines, and ticks
+    tax.boundary(linewidth=1.0)
+    tax.gridlines(color=(0.5, 0.5, 0.5, 0.5), multiple=100/(6*4), linewidth=0.25)
+    tax.gridlines(color=(0.5, 0.5, 0.5, 0.5), multiple=100/6, linewidth=0.5)
+    tax.ticks(axis='b', ticks=[40,50,60,70,80,90,100], linewidth=1, multiple=100/7, tick_formats='%.0f', offset = 0.021, fontsize = 7)
+    tax.ticks(axis='lr', ticks=[0,10,20,30,40,50,60], linewidth=1, multiple=100/7, tick_formats='%.0f', offset = 0.03, fontsize = 7)
+
+    # Set axis labels and title
+    title_fontsize = 9
+    axis_fontsize = 9
+    # tax.set_title("1 Mof Array: BISWEG", fontsize=title_fontsize, y=1.05)
+    # tax.right_corner_label("X", fontsize=fontsize)
+    # tax.top_corner_label("Y", fontsize=fontsize)
+    # tax.left_corner_label("Z", fontsize=fontsize)
+    tax.left_axis_label("Carbon Dioxide", fontsize=axis_fontsize, offset = 0.15)
+    tax.right_axis_label("Oxygen", fontsize=axis_fontsize, offset = 0.15)
+    tax.bottom_axis_label("Nitrogen", fontsize=axis_fontsize, offset = 0.05)
+
+    plt.savefig('/Users/brian_day/Desktop/triplots/%s.png' % array_id, bbox_inches='tight')
```

Comment:
shouldn't have hard code urls in the code base. usually, just saving them to the current directory is good.